jfix-armeria-facade/src/main/kotlin/ru/fix/armeria/facade/webclient/impl/BaseHttpClientBuilderImpl.kt

```diff
@@ -197,9 +176,9 @@ internal abstract class BaseHttpClientBuilderImpl<out HttpClientBuilderT : BaseH
                     }
                     logger.info { "$clientName: closing http client factory under the hood..." }
                     clientFactory.close()
-                    endpointGroup.leftOrNull?.let {
+                    baseBuilderState.endpoint?.leftOrNull?.let {
```

Comment:
Is this OK, that we invoke `baseBuilderState.endpoint.left.close()` - we close endpointGroup which referenced from baseBuilderState?

AutoCloseable refers to concrete WebClient instance, but endpoint  refers to Builder

Reply:
It seems to me the only dangerous place here is `setEndpointGroup(endpointGroup: EndpointGroup)`. By closing endpoint group we are closing the group outside BaseHttpClient. 
But this behaviour is the same as was previously: EndpointGroup object was passed through setEndpointGroup and lambda.
@AnonymousVovus , is that what you are worrying about?
We can go safely and re-create endpoint group or do some deep copy.
@TimurKasatkin , what's your opinion on that?

Reply:
good point from @AnonymousVovus - it creates unexpected mutative behavior - our builders (and their intermediate configurations) must be fully reusable (except for `setEndpointGroup(endpointGroup: EndpointGroup)`)

that is why it has been declared as `endpointGroupCreator` before